### PR TITLE
PTFE-610: bugfix rollback spot instances

### PR DESCRIPTION
**Pull request by Abubakarr99**
Spot instances are still being spawn. This is because the check that obtains the value of the spot key does not work properly. It always returns true. This is a fix.

### CODE REVIEW

**Review by gaspardmoindrot — APPROVED** [review]
LGTM, I test it, it works well

File: srcs/runners_manager/vm_creation/gcloud/GcloudManager.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def configure_instance(
         automatic_restart = True
         provisioning_model = "STANDARD"
         instance_termination_action = "DEFAULT"
-        preemptible = bool(runner.vm_type.config.get("spot", False))
+        preemptible = str(runner.vm_type.config.get("spot", "")).lower() == "true"
```
Comment (tcarmet) [inline srcs/runners_manager/vm_creation/gcloud/GcloudManager.py:123]: I think this is due to how Python's `bool` method behaves, which will consider the following:
```python
>>> bool("false")
True
>>> bool("False")
True
```

Try using the [strtobool](https://docs.python.org/3/distutils/apiref.html#distutils.util.strtobool) method, see if it helps solving our issue.

```suggestion
# assuming the following import is made
from distutils.util import strtobool

        preemptible = strtobool(runner.vm_type.config.get("spot", False))
```

**Review by tcarmet — CHANGES_REQUESTED** [review]

File: srcs/runners_manager/vm_creation/gcloud/GcloudManager.py (hunk 1 above)
Comment (Abubakarr99) [reply srcs/runners_manager/vm_creation/gcloud/GcloudManager.py:123]: yeah this works

Comment (tcarmet) [reply srcs/runners_manager/vm_creation/gcloud/GcloudManager.py:123]: then let's apply it.

**Comment by gaspardmoindrot** [comment]
I tested locally, and it works

**Review by tcarmet — APPROVED** [review]
LGTM, please ensure the code work as expected with `spot: false`

**Comment by gaspardmoindrot** [comment]
> LGTM, please ensure the code work as expected with `spot: false`

Yeah, it works well!

**Pull request merged**